nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/test/java/org/apache/nifi/excel/TestExcelRecordReader.java

```diff
@@ -318,17 +328,47 @@ void testPasswordProtected() throws Exception {
         assertEquals(DATA.length, records.size());
     }
 
+    @Test
+    void testPasswordProtectedOlderExcelFormat() throws Exception {
+        RecordSchema schema = getPasswordProtectedSchema();
+        ExcelRecordReaderConfiguration configuration = new ExcelRecordReaderConfiguration.Builder()
+                .withSchema(schema)
+                .withPassword(PASSWORD)
+                .withAvoidTempFiles(true)
+                .withInputFileType(InputFileType.XLS)
+                .build();
+
+        InputStream inputStream = new ByteArrayInputStream(PASSWORD_PROTECTED_OLDER_EXCEL.toByteArray());
+        ExcelRecordReader recordReader = new ExcelRecordReader(configuration, inputStream, logger);
+        List<Record> records = getRecords(recordReader, false, false);
+
+        assertEquals(DATA.length, records.size());
+    }
```

Comment:
Instead of a new test, combine this test and the previous test line 314-329 into a JUnit 5 org.junit.jupiter.params.ParameterizedTest with an org.junit.jupiter.params.provider.EnumSource. You can use the enum value in the configuration and to choose which password protected workbook to use.

Reply:
Changed